src/segmented_cache_helpers.erl

```diff
@@ -0,0 +1,190 @@
+                           _ -> CurrentIndex - 1
+                       end,
+    case iterate_fun_in_tables(IterativeFun, Key, Segments, Size, LastTableToCheck, CurrentIndex) of
+        {not_found, Value} ->
+            Value;
+        {FoundIndex, Value} ->
+            Strategy = SegmentRecord#segmented_cache.strategy,
+            apply_strategy(Strategy, CurrentIndex, FoundIndex, Key, SegmentRecord),
+            Value
+    end.
+
+-spec iterate_fun_in_tables(IterativeFun, Key, tuple(), Int, Int, Int) ->
+    {not_found, Value} | {non_neg_integer(), Value}
+      when Int :: non_neg_integer(), IterativeFun :: iterative_fun(Key, Value).
+% if we arrived to the last table we finish here
+iterate_fun_in_tables(IterativeFun, Key, Segments, _, LastTableToCheck, LastTableToCheck) ->
+    EtsSegment = element(LastTableToCheck, Segments),
+    case IterativeFun(EtsSegment, Key) of
+        {stop, Value} -> {LastTableToCheck, Value};
+        {continue, Value} -> {not_found, Value}
+    end;
+% if we arrived to the last slot, we check and wrap around
+iterate_fun_in_tables(IterativeFun, Key, Segments, Size, LastTableToCheck, Size) ->
+    EtsSegment = element(Size, Segments),
+    case IterativeFun(EtsSegment, Key) of
+        {stop, Value} -> {Size, Value};
+        {continue, _} -> iterate_fun_in_tables(IterativeFun, Key, Segments, Size, LastTableToCheck, 1)
+    end;
+% else we check the current table and if it fails we move forwards
+iterate_fun_in_tables(IterativeFun, Key, Segments, Size, LastTableToCheck, Index) ->
+    EtsSegment = element(Index, Segments),
+    case IterativeFun(EtsSegment, Key) of
+        {stop, Value} -> {Index, Value};
+        {continue, _} -> iterate_fun_in_tables(IterativeFun, Key, Segments, Size, LastTableToCheck, Index + 1)
+    end.
+
+-spec is_member_fun(ets:tid(), segmented_cache:key()) -> {continue, false} | {stop, true}.
+is_member_fun(EtsSegment, Key) ->
+    case ets:member(EtsSegment, Key) of
+        true -> {stop, true};
+        false -> {continue, false}
+    end.
+
+-spec get_entry_fun(ets:tid(), Key) ->
+    {continue, not_found} | {stop, Value}
+      when Key :: segmented_cache:key(), Value :: segmented_cache:value().
+get_entry_fun(EtsSegment, Key) ->
+    case ets:lookup(EtsSegment, Key) of
+        [{_, Value}] -> {stop, Value};
+        [] -> {continue, not_found}
+    end.
+
+-spec delete_entry_fun(ets:tid(), segmented_cache:key()) -> {continue, true}.
+delete_entry_fun(EtsSegment, Key) ->
+    ets:delete(EtsSegment, Key),
+    {continue, true}.
+
+-spec delete_pattern_fun(ets:tid(), ets:match_pattern()) -> {continue, true}.
+delete_pattern_fun(EtsSegment, Pattern) ->
+    ets:match_delete(EtsSegment, Pattern),
+    {continue, true}.
+
+%% @doc Atomically compare_and_swap an entry, attempt three times, post-check front insert
+-spec put_entry_front(#segmented_cache{}, segmented_cache:key(), segmented_cache:value()) -> boolean().
+put_entry_front(SegmentRecord, Key, Value) ->
```

Comment:
Can be reworked to receive `segmented_cache:name()` as a first parameter. Just to unify if with other interfaces.